processor/aggregationprocessor/processor.go

```diff
@@ -0,0 +1,340 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aggregationprocessor
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"regexp"
+	"sync"
+	"time"
+
+	"github.com/observiq/observiq-otel-collector/processor/aggregationprocessor/internal/aggregate"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatautil"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/multierr"
+	"go.uber.org/zap"
+)
+
+type aggregationProcessor struct {
+	logger   *zap.Logger
+	mux      *sync.Mutex
+	wg       *sync.WaitGroup
```

Comment:
nit: These don't need to be pointers we aren't passing them around in functions so having references to them doesn't help us much.